sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Common/src/Shared/Queues/QueueListener.cs

```diff
@@ -297,10 +338,26 @@ private TaskSeriesCommandResult CreateBackoffResult()
             return new TaskSeriesCommandResult(wait: CreateDelayWithNotificationTask());
         }
 
+        private TaskSeriesCommandResult CreateDelayResult(TimeSpan delay)
+        {
+            Task aggregateTask = Task.WhenAny(_stopWaitingTaskSource.Task, Task.Delay(delay));
+
+            return new TaskSeriesCommandResult(wait: aggregateTask);
+        }
+
         private TaskSeriesCommandResult CreateSucceededResult()
         {
-            Task wait = WaitForNewBatchThreshold();
-            return new TaskSeriesCommandResult(wait);
+            if (_concurrencyManager != null && _concurrencyManager.Enabled)
```

Comment:
In the listener factory we have a null check `_concurrencyManager = concurrencyManager ?? throw new ArgumentNullException(nameof(concurrencyManager));`

Considering that, do we need all the `_concurrencyManager != null` in this file?

Reply:
There are a few paths into the QueueListener ctor. What I'll do is add a null check in the QueueListener ctor (ConcurrencyManager is an optional param currently), and make sure in all cases it'll be non-null. Then I'll remove the other null checks in this file.